core/src/main/java/hudson/security/TokenBasedRememberMeServices2.java

```diff
@@ -142,23 +140,29 @@ public void loginSuccess(HttpServletRequest request, HttpServletResponse respons
 		}
 
 		Assert.notNull(successfulAuthentication.getPrincipal());
-		Assert.notNull(successfulAuthentication.getCredentials());
 		Assert.isInstanceOf(UserDetails.class, successfulAuthentication.getPrincipal());
 
-		long expiryTime = System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(tokenValiditySeconds);
+		long expiryTime = System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(getTokenValiditySeconds());
 		String username = ((UserDetails) successfulAuthentication.getPrincipal()).getUsername();
 
-		String signatureValue = makeTokenSignature(expiryTime, (UserDetails)successfulAuthentication.getPrincipal());
+		String signatureValue = makeTokenSignature(expiryTime, username, ((UserDetails) successfulAuthentication.getPrincipal()).getPassword());
 		String tokenValue = username + ":" + expiryTime + ":" + signatureValue;
 		String tokenValueBase64 = Base64.getEncoder().encodeToString(tokenValue.getBytes());
-		response.addCookie(makeValidCookie(tokenValueBase64, request, tokenValiditySeconds));
+		int tokenLifetime = calculateLoginLifetime(request, successfulAuthentication);
+        /* TODO unclear what the Spring Security equivalent to this is:
+		response.addCookie(makeValidCookie(tokenValueBase64, request, getTokenValiditySeconds()));
+        // something like this, but where is the token signature?
```

Comment:
I don't understand your question about the signature. You have the variable just above, derived from the expiryTime and username.

It's then verified in https://github.com/jenkinsci/jenkins/blob/15ca0323014e6244bb62d598af24b887d90b7323/core/src/main/java/hudson/security/TokenBasedRememberMeServices2.java#L256-L266